packages/grafana-data/src/field/fieldState.test.ts

```diff
@@ -162,14 +157,20 @@ describe('Check field state calculations (displayName and id)', () => {
     const title = checkScenario({
       frames: [
         toDataFrame({
-          fields: [{ name: TIME_SERIES_VALUE_FIELD_NAME }, { name: TIME_SERIES_VALUE_FIELD_NAME }],
+          fields: [
+            { name: TIME_SERIES_VALUE_FIELD_NAME, config: { test: 1 } },
+            { name: TIME_SERIES_VALUE_FIELD_NAME, config: { test: 2 } },
```

Comment:
what is the config?  something seems fishy here -- since i don't know how this is valid (if it fixes something, also seems fishy)

Reply:
Original `===` check was doing strict equality, meaning that the compared fields were the same object in memory while the added `isEqual` is doing deep equality where objects can be the same if they match on deep comparison.

This PR changes the logic to say that the field is the same field either if it passes strict or deep equality. Given this the tests needed to be updated to slightly modify the fields being passed so they were not passing deep equality. For a long time we were doing just strict equality, my PR a few months ago switched this to just deep comparison, and now to get rid of the lodash mock I have switched it to either or which _seems_ to not break anything.

That being said... there is a chance accepting both types of equality for determining if the field is the same may have some strange edge cases 😬

Reply:
all of this was cleaned up by https://github.com/grafana/grafana/pull/81323